pkg/controller/backupscheduled/backupscheduled_controller.go

```diff
@@ -81,95 +104,244 @@ func (r *ReconcileBackupScheduled) Reconcile(request reconcile.Request) (reconci
+	scheduled := &extensionv1.BackupScheduled{}
+	err := r.Get(context.TODO(), request.NamespacedName, scheduled)
 	if err != nil {
-		if kerrors.IsNotFound(err) {
-			return reconcile.Result{}, nil
-		}
-
-		return reconcile.Result{}, err
+		return reconcile.Result{}, errutils.IgnoreNotFound(err)
 	}
 
-	if backupScheduled.Spec.Schedule == "" {
+	if scheduled.Spec.Schedule.CronTab == "" {
 		err := errors.New("BackupScheduled doesn't have a schedule.")
 		log.Error(err.Error())
 		return reconcile.Result{}, err
 	}
 
-	if _, found := backupScheduled.ObjectMeta.GetLabels()["site"]; !found {
+	if _, found := scheduled.ObjectMeta.GetLabels()["site"]; !found {
 		err := errors.New("BackupScheduled doesn't have a site label.")
 		log.Error(err.Error())
 		return reconcile.Result{}, err
 	}
 
-	log.Info("Calculating next scheduled backup.")
-	now := time.Now()
-	status := extensionv1.BackupScheduledStatus{
-		LastExecutedTime: &metav1.Time{Time: getScheduleComparison(backupScheduled.Status, now)},
+	log.Info("Querying Backups")
+	var backups extensionv1.BackupList
+	listOptions := client.MatchingField(OwnerKey, request.Name)
+	listOptions.Namespace = request.Namespace
+	err = r.List(context.Background(), listOptions, &backups)
+	if err != nil {
+		return reconcile.Result{}, err
 	}
 
-	// Check if we are currently due for a backup to be scheduled.
-	expr, err := cronexpr.Parse(backupScheduled.Spec.Schedule)
+	log.Info("Filtering Backups")
+	active, successful, failed, err := r.SortBackups(scheduled, backups)
 	if err != nil {
-		log.Errorf("Cron schedule syntax is invalid: '%s'.", err.Error())
 		return reconcile.Result{}, err
 	}
-	next := expr.Next(status.LastExecutedTime.Time)
-	if next.Before(now) || next.Equal(now) {
-		// Due for a backup - create object.
-		log.Info("Backup due - creating.")
-		backup := &extensionv1.Backup{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      fmt.Sprintf("%s-%d", backupScheduled.Name, now.Unix()),
-				Namespace: backupScheduled.Namespace,
-				Labels:    backupScheduled.Labels,
-				Annotations: map[string]string{
-					restic.FriendlyNameAnnotation: now.Format("Mon, 02/01/2006 - 15:04"),
-				},
-			},
-			Spec: extensionv1.BackupSpec{
-				MySQL:   backupScheduled.Spec.MySQL,
-				Volumes: backupScheduled.Spec.Volumes,
-			},
+
+	log.Infof("Job count: active=%d success=%d failed=%d", len(active), len(successful), len(failed))
+
+	err = r.Status().Update(context.TODO(), scheduled)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	log.Info("Cleaning up old Backups")
+	err = r.Cleanup(scheduled, successful, failed)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if scheduled.Spec.Schedule.Suspend != nil && *scheduled.Spec.Schedule.Suspend {
+		log.Info("Scheduling has been suspended. Skipping.")
+		return reconcile.Result{}, nil
+	}
+
+	return r.ScheduleNextBackup(scheduled, active)
+}
+
+func getScheduleComparison(s shpmetav1.ScheduledStatus, now time.Time) time.Time {
```

Comment:
Doesn't look like this is used anywhere?

Reply:
Removed! thanks spotting it 👍